Sources/Basic/OrderedSet.swift

```diff
@@ -77,6 +77,39 @@ public struct OrderedSet<E: Hashable>: Equatable, Collection {
         array.removeAll(keepingCapacity: keepCapacity)
         set.removeAll(keepingCapacity: keepCapacity)
     }
+
+    // MARK:- MutableCollection, RandomAccessCollection conformance
+
+    public var startIndex: Int { return contents.startIndex }
+    public var endIndex: Int { return contents.endIndex }
+
+    public subscript(position: Int) -> Element {
+        get {
+            return array[position]
+        }
+        set(newValue) {
+            let oldValue = array[position]
+            // Remove the old value from set.
+            set.remove(oldValue)
+            // Add the new value.
+            array[position] = newValue
```

Comment:
This can leave the array in an inconsistent state if the new value being inserted is already present in the set. I'm not sure exactly what the semantics should be in that case, I would probably suggestion we make it a `preconditionFailure` for now.